src/t_list.c

```diff
@@ -1001,3 +1121,69 @@ void brpoplpushCommand(client *c) {
         != C_OK) return;
     blmoveGenericCommand(c, LIST_TAIL, LIST_HEAD, timeout);
 }
+
+/* LPOP takes one key, and can return multiple elements.
+ * BLPOP takes multiple keys, but returns one element from just one key.
+ * [B]LMPOP can take multiple keys and return multiple elements from just one key.
+ *
+ * LMPOP numkeys [<key> ...] LEFT|RIGHT [COUNT count] [BLOCK timeout]
+ *
+ * [B]LMPOP
+ * 1) keyname
+ * 2) 1) element1
+ *    2) element2 */
+void lmpopCommand(client *c) {
+    int j;
+    int key_start = 2;     /* Skip the command and numkeys. */
+    long numkeys = 0;      /* Number of keys. */
+    int where = -1;        /* HEAD for LEFT, TAIL for RIGHT. */
+    int is_left = 0;       /* Pop from the beginning of the list. */
+    int is_right = 0;      /* Pop from the end of the list. */
+    long count = 1;        /* Reply will consist of up to count elements, depending on the list's length. */
+    int is_block = 0;      /* Set to 1 if we are in BLOCK variant. */
+    int timeout_idx = 0;   /* The index of timeout in the c->argv. */
+
+    if (getRangeLongFromObjectOrReply(c, c->argv[1], 1,
+            LONG_MAX, &numkeys, "numkeys should be greater than 0") != C_OK)
+        return;
+
+    /* Parse the optional arguments first. */
+    for (j = key_start + numkeys; j < c->argc; j++) {
+        char *opt = c->argv[j]->ptr;
+        int moreargs = (c->argc - 1) - j;
+
+        if (!strcasecmp(opt, "LEFT") && !is_right) {
+            is_left = 1;
+            where = LIST_HEAD;
+        } else if (!strcasecmp(opt, "RIGHT") && !is_left) {
+            is_right = 1;
+            where = LIST_TAIL;
```

Comment:
i think we meant that LEFT or RIGHT are positional arguments (we always know their position).
so this loop should only search for COUNT and BLOCK  (both optional)

Reply:
right. alerady modified it